delivery_free_fee_removal/tests/test_delivery_free_fee_removal.py

```diff
@@ -58,3 +80,25 @@ def test_delivery_free_fee_removal_free_fee(self):
         self.assertRecordValues(
             delivery_line, [{"is_free_delivery": True, "qty_to_invoice": 0}]
         )
+        self.deliver_and_invoice_order(self.sale)
+
+    def deliver_and_invoice_order(self, order):
+        """Check delivery line state for invoicing and delivery."""
+        # Deliver the order
+        picking = order.picking_ids
+        self.assertEqual(picking.state, "assigned")
+        picking.move_line_ids[0].qty_done = 1.0
+        picking.action_done()
```

Comment:
do you actually need to deliver the picking for the test?

Reply:
I guess not, I wrote the test to understand what was required. They do look over the top.

Reply:
Removing them